### ISSUE TITLE
Smooth Normalizing Flow N dimensions

### ISSUE THREAD

**Issue by Justinezgh** [astrodeepnet/sbi_experiments#23]
Finally works for more than 2d :) 
I changed the bijector and the NN in the NF and tested it on 6d: [notebook](https://colab.research.google.com/drive/1YSjUofeeAhCqxb4L2tkiyInZFCxD-x27?usp=sharing)

**Comment by Justinezgh** [comment]
the bijector works :) 
I learned a 6d gaussian, green is the truth and blue the prediction. 

learned with nll:
![image](https://user-images.githubusercontent.com/72011736/156606984-e4092ee5-0ca7-48f4-8398-273c0ca719b9.png)

learned with score matching to check the score:
![image](https://user-images.githubusercontent.com/72011736/156655351-d7afd979-2f21-4687-aaad-bfd1fc7aa822.png)

[notebook](https://colab.research.google.com/drive/1YSjUofeeAhCqxb4L2tkiyInZFCxD-x27?usp=sharingl)

**Comment by Justinezgh** [comment]
I tested a first 6d gaussian model (without latent variables):

```python 
@jax.jit
def get_batch(seed):

  prior = tfd.Independent(tfd.Uniform(0.1*jnp.ones(d), 0.9*jnp.ones(d)),
                            reinterpreted_batch_ndims=1)
  
  mu = prior.sample(batch_size, seed=seed)
  
  batch = jax.vmap(lambda mu, seed: tfd.Independent(tfd.TruncatedNormal(mu,
                                                                        0.08*jnp.ones(d),
                                                                        0.01,
                                                                        0.99),
                                                    1).sample(seed=seed))(mu,jax.random.split(seed, batch_size))
  score = jax.vmap(jax.grad(lambda batch, mu: tfd.Independent(tfd.Normal(mu,0.08*jnp.ones(d)),1).log_prob(batch)))(batch, mu)
  score += jax.vmap(jax.grad(prior.log_prob))(mu)

  return mu, batch, score
```

learned without score: (the red point is the true mean)
![image](https://user-images.githubusercontent.com/72011736/156802782-04d100cd-c6cd-4889-87dc-2672cdb6c242.png)

learned with loss = 1e-3 * score - nll:
![image](https://user-images.githubusercontent.com/72011736/156802959-8b6c4ef7-b63f-4102-a3ed-5c6135656780.png)
